lib/vagrant-service-manager/service.rb

```diff
@@ -0,0 +1,35 @@
+# Loads all services
+Dir["#{File.dirname(__FILE__)}/services/*.rb"].each { |f| require_relative f }
+
+module Vagrant
+  module ServiceManager
+    SUPPORTED_BOXES = ['adb', 'cdk']
+
+    class Service
+      def initialize(app, env)
+        @app = app
+        @machine = env[:machine]
+        @ui = env[:ui]
+        @services = @machine.config.servicemanager.services.split(',').map(&:strip)
+        @docker_hook = Docker.new(@machine, @ui)
+        @openshift_hook = OpenShift.new(@machine, @ui)
+      end
+
+      def call(env)
+        @app.call(env)
+
+        if SUPPORTED_BOXES.include? @machine.guest.capability(:os_variant)
+          @docker_hook.execute
+
+          if @machine.guest.capability(:os_variant) == "cdk"
```

Comment:
I understand it to be like this

If no service line and ADB - execute docker
if no service line and CDK - execute openshift
if service line execute what is on service line
otherwise do nothing